lib/streamlit/delta_generator.py

```diff
@@ -355,16 +356,41 @@ def wrapper(*args: Any, **kwargs: Any) -> NoReturn:
         return wrapper
 
     @property
-    def _parent_block_types(self) -> Iterable[str]:
+    def _parent_block_types(self) -> Iterable[Tuple[str, Optional[int]]]:
         """Iterate all the block types used by this DeltaGenerator and all
         its ancestor DeltaGenerators.
         """
         current_dg: Optional[DeltaGenerator] = self
         while current_dg is not None:
             if current_dg._block_type is not None:
-                yield f"{current_dg._block_type}:{current_dg._root_container}"
+                yield current_dg._block_type, current_dg._root_container
             current_dg = current_dg._parent
 
+    def _get_raw_parent_block_types(
+        self, parent_block_types: Iterable[Tuple[str, Optional[int]]]
+    ) -> FrozenSet[str]:
+        parent_block_types_raw = []
+        for parent_block, _ in parent_block_types:
+            parent_block_types_raw.append(parent_block)
+        return frozenset(parent_block_types_raw)
+
+    def _count_num_of_parent_columns(
+        self, parent_block_types: Iterable[Tuple[str, Optional[int]]]
+    ) -> int:
+        num_of_columns = 0
+        for parent_block, _ in parent_block_types:
+            if "column" == parent_block:
+                num_of_columns = num_of_columns + 1
+        return num_of_columns
+
+    def _check_if_one_of_parents_is_a_sidebar(
+        self, parent_block_types: Iterable[Tuple[str, Optional[int]]]
+    ) -> bool:
+        for _, root_container in parent_block_types:
+            if RootContainer.SIDEBAR == root_container:
```

Comment:
this can be collapsed into `return RootContainer.SIDEBAR == root_container`

Reply:
@sfc-gh-mnowotka are you sure it can be? Maybe I misunderstood you, but:

This is a `for` loop, so if I collapse it to
```
        for _, root_container in parent_block_types:
            return RootContainer.SIDEBAR == root_container:
```
it will end immediately checking only first element and I need to check if there is at least 1 occurence in whole Iterable.
If I misunderstood can you please explain it to me differently? or let's make a call

For now I'm leaving it as it is, since I think it's not going to work as expected if I use `return` here

Reply:
What do you think about using `any`?
```
return any(
    RootContainer.SIDEBAR == root_container
    for _, root_container in parent_block_types:
)
```

Reply:
Thank you @sfc-gh-kbregula 🙇 , this is really cool, now I can see how can we use `return` here, thank you @sfc-gh-mnowotka 🙇 The PR is updated 👍

Reply:
Sorry Tomek, I read your code too fast. My advice wasn't usable but Kamil's idea will `all` was a great one :)